src/pokemonClass.ts

```diff
@@ -0,0 +1,74 @@
+export type pokemonType = 'grass' | 'poison' | 'fire' | 'water' | 'bug' | 'normal' | 'flying' | 'electric' | 'ground' | 'fairy' | 'psychic' | 'fighting' | 'rock' | 'ground'
+
+export abstract class Pokemon {
+    constructor() {}
+    abstract _image: string;
+    abstract _HP: number;
```

Comment:
it needs two separate values: "max hp" (that will be constant) and "current hp" (that can be in range between 0 and  max)